### PR TITLE
 openfortivpn: fix on darwin

### PR DESCRIPTION
**Pull request by spacelatte**
make ppp optional since it's already provided by the system

###### Motivation for this change

- `openfortivpn` is not installable on Darwin due to `ppp` isn't available on the platform.
- The `ppp` is already provided by the macOS itself...

image

###### Things done

- Makes `ppp` optional dependency. Used in Linux but not Darwin

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/openfortivpn/default.nix [hunk 1]
```diff
@@ -1,9 +1,12 @@
 { stdenv, lib, fetchFromGitHub, autoreconfHook, pkg-config
-, openssl, ppp
-, systemd ? null }:
+, openssl
+, ppp
+, systemd
+}:
 
 let
   withSystemd = stdenv.isLinux && !(systemd == null);
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openfortivpn/default.nix:8]:
```suggestion
  withSystemd = stdenv.isLinux;
```

File: pkgs/tools/networking/openfortivpn/default.nix [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
-, systemd ? null }:
+, openssl
+, ppp
+, systemd
+}:
 
 let
-  withSystemd = stdenv.isLinux && !(systemd == null);
+  withSystemd = stdenv.isLinux;
+  withPpp     = stdenv.isLinux;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openfortivpn/default.nix:9]: Normally we expose those options as inputs for convince.

**Comment by spacelatte** [comment]
hi, how can i make this as input? (total noob here)

File: pkgs/tools/networking/openfortivpn/default.nix (hunk 2 above)
Comment (spacelatte) [reply pkgs/tools/networking/openfortivpn/default.nix:9]: how can I make a such thing? can you show me an example?

File: pkgs/tools/networking/openfortivpn/default.nix [hunk 3]
```diff
@@ -1,9 +1,12 @@
 { stdenv, lib, fetchFromGitHub, autoreconfHook, pkg-config
-, openssl, ppp
-, systemd ? null }:
+, openssl
+, ppp
+, systemd
+}:
 
 let
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openfortivpn/default.nix:7]:
```suggestion
, withSystemd = stdenv.isLinux;
, withPpp = stdenv.isLinux;
}:

```

File: pkgs/tools/networking/openfortivpn/default.nix [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
-  withSystemd = stdenv.isLinux && !(systemd == null);
+  withSystemd = stdenv.isLinux;
+  withPpp     = stdenv.isLinux;
 
 in
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openfortivpn/default.nix:12]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/networking/openfortivpn/default.nix [hunk 5]
```diff
@@ -1,11 +1,11 @@
 { stdenv, lib, fetchFromGitHub, autoreconfHook, pkg-config
-, openssl, ppp
-, systemd ? null }:
+, openssl
+, ppp
+, systemd
+, withSystemd = stdenv.isLinux;
+, withPpp = stdenv.isLinux;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openfortivpn/default.nix:6]:
```suggestion
, withSystemd ? stdenv.isLinux
, withPpp ? stdenv.isLinux
```
total derp of me

Comment (spacelatte) [reply pkgs/tools/networking/openfortivpn/default.nix]: honestly, i have no idea about nix (language)

just switched my macos over to nix (from my own set of bash scripts) to manage packages.

noticed this gem, tried with some tutorial changes, wanted to contribute.

thank you for your help and patience!

**Pull request merged**